service/qos/service_level_controller.cc

```diff
@@ -384,6 +385,14 @@ future<> service_level_controller::notify_service_level_removed(sstring name) {
     co_return;
 }
 
+future<> service_level_controller::notify_effective_service_levels_cache_reloaded() {
+    return seastar::async([this] {
+        _subscribers.thread_for_each([] (qos_configuration_change_subscriber* subscriber) {
+            subscriber->on_effective_service_levels_cache_reloaded().get();
+        });
+    });
+}
+
```

Comment:
I'm not sure what we gain here by using `seastar::async` and `thread_for_each`. The `on_effective_service_levels_cache_reloaded` method that is called in the loop returns a future, so it is not assume presence of `seastar::async` and does not take advantage of it. `seastar::async` allocates a 128kb stack, so using it is a bit wasteful here.

I suggest removing `seastar::async` and changing this to `_subscriber.for_each`.